pkg/iac/terraform/state/terraform_state_reader.go

```diff
@@ -220,6 +222,9 @@ func (r *TerraformStateReader) retrieveMultiplesStates() ([]resource.Resource, e
 func read(path string, reader backend.Backend) (*states.State, error) {
 	state, err := readState(path, reader)
 	if err != nil {
+		if _, ok := reader.(*backend.HTTPBackend); ok && strings.Contains(err.Error(), "The state file could not be parsed as JSON") {
+			return nil, errors.Errorf("given url is not a valid statefile")
```

Comment:
I'm confused with that change as it's less explicit than the current error : 

```
2 problems:

- Unsupported state file format: The state file could not be parsed as JSON: syntax error at byte offset 1.
- Unsupported state file format: The state file does not have a "version" attribute, which is required to identify the format version.
```

I don't understand how it'll solve the original issue, can you elaborate on that? thanks